scripts/lib/links/markdown.ts

```diff
@@ -0,0 +1,85 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2023.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import { readFile } from "fs/promises";
+import path from "node:path";
+
+import markdownLinkExtractor from "markdown-link-extractor";
+import { visit } from "unist-util-visit";
+import { unified } from "unified";
+import remarkStringify from "remark-stringify";
+import { Root } from "remark-mdx";
+import rehypeRemark from "rehype-remark";
+import rehypeParse from "rehype-parse";
+import remarkGfm from "remark-gfm";
+
+interface JupyterCell {
+  cell_type: string;
+  source: string[];
+}
+
+export function markdownFromNotebook(rawContent: string): string {
+  const cells = JSON.parse(rawContent).cells as JupyterCell[];
+  return cells
+    .filter((cell) => cell.cell_type === "markdown")
+    .map((cell) => cell.source.join(""))
+    .join("\n");
+}
+
+export function parseAnchors(markdown: string): string[] {
+  const result = markdownLinkExtractor(markdown).anchors;
+  const idAnchors = markdown.match(/(?<=id=")(.*)(?=")/gm);
+  if (idAnchors !== null) {
+    result.push(...idAnchors.map((id) => `#${id}`));
+  }
+  return result;
+}
```

Comment:
I like this new function, it makes the code simpler and more readable. Good idea with the spread operator! Moreover, we could also get rid of the if condition and the push by using the spread operator in the return. I still like the way you defined the function though, just FYI.

```suggestion
export function parseAnchors(markdown: string): string[] {
  const mdAnchors = markdownLinkExtractor(markdown).anchors;
  const idAnchors = markdown.match(/(?<=id=")(.*)(?=")/gm) || [];
  return [...mdAnchors, ...idAnchors.map((id) => `#${id}`)];
}
```

Reply:
I love it 😍 Thanks!